rgw/v2/tests/s3_swift/test_lc_reshard.py

```diff
@@ -0,0 +1,153 @@
+-Run "radosgw-admin lc reshard fix" should create an entry for removed LC shards
+-Remove the user at successful completion
+"""
+import os
+import sys
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+import argparse
+import json
+import logging
+import time
+import traceback
+
+import v2.lib.resource_op as s3lib
+import v2.utils.utils as utils
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.resource_op import Config
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.s3.auth import Auth
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.tests.s3_swift import reusable
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+from v2.utils.utils import HttpResponseParser, RGWService
+
+log = logging.getLogger(__name__)
+TEST_DATA_PATH = None
+
+
+def test_exec(config, ssh_con):
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    ceph_conf = CephConfOp(ssh_con)
+    rgw_service = RGWService()
+
+    # create user
+    all_users_info = s3lib.create_users(config.user_count)
+    for each_user in all_users_info:
+        # authenticate
+        auth = Auth(each_user, ssh_con, ssl=config.ssl)
+        rgw_conn = auth.do_auth()
+        rgw_conn2 = auth.do_auth_using_client()
+        # create buckets
+        if config.test_ops["create_bucket"] is True:
+            log.info("no of buckets to create: %s" % config.bucket_count)
+            for bc in range(config.bucket_count):
+                bucket_name = utils.gen_bucket_name_from_userid(
+                    each_user["user_id"], rand_no=bc
+                )
+                bucket = reusable.create_bucket(bucket_name, rgw_conn, each_user)
+                life_cycle_rule = {"Rules": config.lifecycle_conf}
+                reusable.put_bucket_lifecycle(
+                    bucket, rgw_conn, rgw_conn2, life_cycle_rule
+                )
+
+    log.info("Checking the LC shard IDs assigned to all the buckets")
+    lc_data = json.loads(utils.exec_shell_cmd("radosgw-admin lc list"))
+    shard_data = []
+    for shard_num in lc_data:
+        shard_data.append(shard_num["shard"])
+
+    # Check whether there are 32 lc shards
+    numlc = utils.exec_shell_cmd("rados ls -p default.rgw.log -N lc | wc -l")
+    print(numlc)
+    if int(numlc.strip()) != 32:
+        raise TestExecError("Number of LC shards not 32 by default")
+
+    log.info("Remove the LC shards currently assigned to buckets")
+    for data in shard_data:
+        op = utils.exec_shell_cmd(f"rados rm -p default.rgw.log -N lc {data}")
+
+    # Check there are less than 32 LC shards
+    numlc = utils.exec_shell_cmd("rados ls -p default.rgw.log -N lc | wc -l")
+    if int(numlc.strip()) >= 32:
+        raise TestExecError("LC shards not deleted")
```

Comment:
@TejasC88 , could we create an exception in the exceptions.py and use it for the lc exceptions